pkg/apis/core/validation/shoot.go

```diff
@@ -865,6 +865,19 @@ func validateKubernetes(kubernetes core.Kubernetes, dockerConfigured, shootHasDe
 			}
 		}
 
+		if kubeAPIServer.AccessControl != nil {
+			// Action available, but no Source
+			if kubeAPIServer.AccessControl.Action != nil && kubeAPIServer.AccessControl.Source == nil {
+				allErrs = append(allErrs, field.Required(fldPath.Child("kubeAPIServer", "accessControl", "action"), "is required when access control action enabled"))
+			}
+			// Source available, but no Action
+			if kubeAPIServer.AccessControl.Source != nil {
+				if kubeAPIServer.AccessControl.Action == nil {
+					allErrs = append(allErrs, field.Required(fldPath.Child("kubeAPIServer", "accessControl", "source"), "is required when source for action enabled"))
```

Comment:
Same as above, but just reversed.
```suggestion
					allErrs = append(allErrs, field.Required(fldPath.Child("kubeAPIServer", "accessControl", "action"), "is required when source for action enabled"))
```